### PR TITLE
HPCC-22241 Remove legacy authentication code

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jakesmith** [comment]
@mckellyln - please review

File: fs/dafsclient/rmtfile.hpp [hunk 1]
```diff
@@ -66,8 +66,6 @@ extern DAFSCLIENT_API void disconnectRemoteIoOnExit(IFileIO *fileio,bool set=tru
 extern DAFSCLIENT_API bool resetRemoteFilename(IFile *file, const char *newname); // returns false if not remote
 
 
-extern DAFSCLIENT_API void enableAuthentication(bool set=true); // default enabled for clients, disabled for server
-
```
Comment (mckellyln) [inline fs/dafsclient/rmtfile.hpp]: ok to remove entry point ?  Should it be a no-op ?  Does any client link to this lib that we should be concerned with ?

**Comment by mckellyln** [comment]
@jakesmith looks fine, but one small question.

**Comment by mckellyln** [comment]
@jakesmith I am ok with removing the entry point, just wanted to ask about it.
Looks like this needs a rebase though.
ok to merge after rebase.

File: fs/dafsclient/rmtfile.hpp (hunk 1 above)
Comment (jakesmith) [reply fs/dafsclient/rmtfile.hpp]: I think fine to remove, only other (same version) platform dlls link to this.

**Comment by jakesmith** [comment]
@mckellyln - thanks. Based/resolved conflict. Please check.

**Comment by mckellyln** [comment]
@jakesmith looks good to merge.

**Comment by jakesmith** [comment]
@ghalliday - please merge.

**Pull request merged**